### ISSUE TITLE
Feature request: Add mode argument to discriminator function in tfgans GANEstimator

### ISSUE THREAD

**Issue by ptats** [tensorflow/tensorflow#17082]
Can we add mode as an optional argument to the discriminator function in the tfgan GANEstimator? This would allow the discriminator function to perform batch normalisation and dropout.
Posted this on stack overflow:

https://stackoverflow.com/questions/48837537/why-cant-i-pass-mode-to-the-discriminator-function-in-tensorflows-ganestimator

**Comment by ptats** [comment]
Sorry just saw this feature has been completed
